### PR TITLE
Fix temp path handling on Windows

### PR DESCRIPTION
**Pull request by mbland**
Follow up to #17.

On Windows, Dir.tmpdir may returned shorted versions of temp dir components, e.g. "C:/Users/JOHNDO~1/AppData/Local/Temp" for "C:/Users/John Doe/AppData/Local/Temp".

The previous implementation of Compiler.tmp_dir from 91b7ead used File.expand_path, but this only expands a shortened path component if it's the last component in the path. The test now reproduces the problem on Windows, and verifies the fix, which uses Dir.glob instead.

### CODE REVIEW

**Comment by luislavena** [comment]
Thank you @mbland for your patience (after this long silence on my end) and taking the time to document the issue.

What if we use `File.realpath` instead of `Dir.glob`? That one not only should ensure all the components of the given path are expanded but also they all exist.

`File.realpath` is present in Ruby 1.9.3 and greater. Since I dropped support for anything older than 2.0.0, I think is safe.

I no longer have a Windows machine to check this out, but will appreciate if you can try and let me know.

Thank you.

**Comment by mbland** [comment]
No dice; `File.realpath` causes the same failure as before.

**Comment by luislavena** [comment]
@mbland thank you for taking the test, really shame that Ruby is not consistent on that :cry: 

I think will be best to skip the test on anything other than Windows since gem-compiler itself relies on RubyGems build extension mechanism which can handle path with spaces.

Will add an extra comment shortly.

Thank you once again.

File: test/rubygems/test_gem_compiler.rb [hunk 1]
```diff
@@ -109,6 +109,10 @@ def test_compile_bundle_artifacts
     assert_includes spec.files, "lib/#{artifact}"
   end
 
+  # We need to check that tempdir paths that contain spaces as are handled
+  # properly on Windows. In some cases, Dir.tmpdir may returned shorted
+  # versions of these components, e.g.  "C:/Users/JOHNDO~1/AppData/Local/Temp"
+  # for "C:/Users/John Doe/AppData/Local/Temp".
   def test_compile_bundle_artifacts_path_with_spaces
```
Comment (luislavena) [inline test/rubygems/test_gem_compiler.rb]: Let's `skip` the test unless platform is Windows.

Comment (mbland) [inline test/rubygems/test_gem_compiler.rb]: Done.

**Comment by luislavena** [comment]
Thank you @mbland, merged!

**Comment by mbland** [comment]
Thanks, @luislavena!

**Pull request merged**